ui/src/logs/containers/LogsPage.tsx

```diff
@@ -74,34 +100,49 @@ class LogsPage extends PureComponent<Props> {
 
   private handleChooseTimerange = (timeRange: TimeRange) => {
     this.props.setTimeRange(timeRange)
+    this.props.executeHistogramQuery()
   }
 
   private handleChooseSource = (sourceID: string) => {
     this.props.getSource(sourceID)
   }
 
   private handleChooseNamespace = (namespace: Namespace) => {
-    // Do flip
     this.props.setNamespace(namespace)
   }
+
+  private handleChartZoom = (lower, upper) => {
+    if (lower) {
+      this.props.changeZoom({lower, upper})
+    }
+  }
 }
 
 const mapStateToProps = ({
   sources,
-  logs: {currentSource, currentNamespaces, timeRange, currentNamespace},
+  logs: {
+    currentSource,
+    currentNamespaces,
+    timeRange,
+    currentNamespace,
+    histogramData,
+  },
 }) => ({
   sources,
   currentSource,
   currentNamespaces,
   timeRange,
   currentNamespace,
+  histogramData,
 })
 
 const mapDispatchToProps = {
-  getSource: getSourceAsync,
+  getSource: getSourceAndPopulateNamespaces,
```

Comment:
so have we been adding a bunch of unnecessary boilerplate by making `mapDispatchToProps` a function with all of the `bindActionCreators` stuff?